src/main/java/com/capgemini/wsb/fitnesstracker/training/api/TrainingProvider.java

```diff
@@ -13,6 +16,13 @@ public interface TrainingProvider {
      * @param trainingId id of the training to be searched
      * @return An {@link Optional} containing the located Training, or {@link Optional#empty()} if not found
      */
-    Optional<User> getTraining(Long trainingId);
+    Optional<Training> getTrainingById(Long trainingId);
 
+    List<TrainingDto> findAllTrainings();
+
+    List<Training> getTrainingsByUser(UserDto user);
+
+    Training createTraining(Training training);
+
+    List<TrainingDto> getTrainingsByActivityType(ActivityType activityType);
```

Comment:
Consider renaming method to something like `findTrainingsByActivityTypeByUser`. In Java get is used mostly for getters.